src/Uno.Extensions.Navigation.UI/ThemeService.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Uno.Toolkit.UI;
+using Windows.Storage;
+
+namespace Uno.Extensions.Navigation.UI
+{
+	public class ThemeService : IThemeService
+	{
+		private readonly Window _window;
+		private readonly IDispatcher _dispatcher;
+		private ApplicationTheme _storedTheme;
+		private ApplicationDataContainer _localSettings;
+		private const string _storedThemeKey = "localStorageThemeKey";
+		private const string _useSystemThemeKey = "localStorageSystemThemeKey";
+
+		public ThemeService(Window window, IDispatcher dispatcher)
+		{
+			_window = window;
+			_dispatcher = dispatcher;
+			_storedTheme = SystemThemeHelper.GetCurrentOsTheme();
```

Comment:
Take a look at LocalizationService as an example of how to persist values - you shouldn't have to do write all this logic for persisting state. Use IWritableOptions